bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/ComfoAirHandler.java

```diff
@@ -0,0 +1,308 @@
+                        } else if (dataType instanceof DataTypeNumber || dataType instanceof DataTypeRPM) {
+                            state = (DecimalType) command;
+                        } else if (dataType instanceof DataTypeTemperature) {
+                            if (command instanceof QuantityType<?>) {
+                                QuantityType<?> celsius = ((QuantityType<?>) command).toUnit(SIUnits.CELSIUS);
+                                if (celsius != null) {
+                                    state = new DecimalType(celsius.doubleValue());
+                                }
+                            } else {
+                                state = (DecimalType) command;
+                            }
+                        } else if (dataType instanceof DataTypeVolt) {
+                            if (command instanceof QuantityType<?>) {
+                                QuantityType<?> volts = ((QuantityType<?>) command).toUnit(SmartHomeUnits.VOLT);
+                                if (volts != null) {
+                                    state = new DecimalType(volts.doubleValue());
+                                }
+                            } else {
+                                state = (DecimalType) command;
+                            }
+                        }
+                    }
+                }
+
+                if (state != null) {
+                    ComfoAirCommand changeCommand = ComfoAirCommandType.getChangeCommand(channelId, state);
+
+                    if (changeCommand != null) {
+                        sendCommand(changeCommand, channelId);
+
+                        Collection<ComfoAirCommand> affectedReadCommands = ComfoAirCommandType
+                                .getAffectedReadCommands(channelId, keysToUpdate);
+
+                        if (affectedReadCommands.size() > 0) {
+                            Runnable updateThread = new AffectedItemsUpdateThread(affectedReadCommands);
+                            scheduler.schedule(updateThread, 3, TimeUnit.SECONDS);
+                        }
+                    }
+                } else {
+                    logger.warn("Unhandled command type: {}", command.toString());
+                }
+            } catch (final RuntimeException e) {
+                logger.warn("Updating ComfoAir failed: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ComfoAirConfiguration config = getConfigAs(ComfoAirConfiguration.class);
+        String serialPort = config.serialPort;
+
+        if (StringUtils.isNotEmpty(serialPort)) {
+            comfoAirConnector = new ComfoAirSerialConnector(serialPortManager, serialPort, BAUDRATE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            return;
+        }
```

Comment:
It is better to arrange your logic in a fail-fast fashion. Also, caching the field as a local variable will allow you to only have to do null checks once. Almost all the null check warnings regarding class fields can be solved by making a local variable to cache the field's value.
```suggestion
        if (StringUtils.isEmpty(serialPort)) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
            return;
        }
        ComfoAirSerialConnector comfoAirConnector = new ComfoAirSerialConnector(serialPortManager, serialPort, BAUDRATE);
        this.comfoAirConnector = comfoAirConnector;
```